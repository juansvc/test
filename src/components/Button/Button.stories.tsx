import React from 'react';

import { Meta,Story } from '@storybook/react';
import { ThemeProvider } from '@zenprojects/ndp-design-system';

import { Button } from './Button';

export default {
  title: 'Example/Button',
  component: Button,
  argTypes: {},
} as Meta;

const Template: Story = () => (
  <ThemeProvider>
    <Button />
  </ThemeProvider>
);

export const Primary = Template.bind({});
