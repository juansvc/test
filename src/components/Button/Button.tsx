import { Button as NDPButton } from '@zenprojects/ndp-design-system';

export const Button = ({ children }: { children: React.ReactNode }): JSX.Element => {
  return <NDPButton>{children}</NDPButton>;
};
