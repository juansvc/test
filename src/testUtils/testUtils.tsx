import React, { ReactElement } from 'react';

import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@zenprojects/ndp-design-system';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AllTheProviders: React.FC = ({ children }: React.PropsWithChildren<Record<string, any>>) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender: any = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
