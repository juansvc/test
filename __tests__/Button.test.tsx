import { Button } from '../src/components/Button/Button';
import { render } from '../src/testUtils/testUtils';

describe('Button', () => {
  it('should render the Button component with the default props', () => {
    const container = render(<Button>I am a button</Button>);

    expect(container).toMatchSnapshot();
  });
});
