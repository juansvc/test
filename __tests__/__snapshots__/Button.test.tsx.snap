// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the Button component with the default props 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  display: inline-grid;
  grid-gap: calc(((7 / 16) * 1rem) + (9.335 - 7) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  grid-auto-flow: column;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 200ms background-color ease-out;
  transition: 200ms background-color ease-out;
  cursor: pointer;
  border-radius: calc(((7 / 16) * 1rem) + (9.335 - 7) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  font-size: clamp(0.88rem,calc(0.83rem + 0.21vw),1.17rem);
  font-weight: 700;
  font-family: 'GT-America Standard Bold',sans-serif;
  vertical-align: middle;
  color: #1A1A1A;
  padding: calc(((10.5 / 16) * 1rem) + (14.002500000000001 - 10.5) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320))) calc(((14 / 16) * 1rem) + (18.67 - 14) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  background: #F15A28;
  border: 1px solid transparent;
}

.c0 svg {
  height: calc(((10.5 / 16) * 1rem) + (14.002500000000001 - 10.5) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  width: calc(((14 / 16) * 1rem) + (18.67 - 14) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
}

.c0:hover {
  background: #F26B3E;
  border: 1px solid transparent;
}

.c0:active {
  background: #F15A28;
  color: #FFFFFF;
}

<body>
    <div>
      <button
        class="c0"
      >
        I am a button
      </button>
    </div>
  </body>,
  "container": .c0 {
  display: inline-grid;
  grid-gap: calc(((7 / 16) * 1rem) + (9.335 - 7) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  grid-auto-flow: column;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 200ms background-color ease-out;
  transition: 200ms background-color ease-out;
  cursor: pointer;
  border-radius: calc(((7 / 16) * 1rem) + (9.335 - 7) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  font-size: clamp(0.88rem,calc(0.83rem + 0.21vw),1.17rem);
  font-weight: 700;
  font-family: 'GT-America Standard Bold',sans-serif;
  vertical-align: middle;
  color: #1A1A1A;
  padding: calc(((10.5 / 16) * 1rem) + (14.002500000000001 - 10.5) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320))) calc(((14 / 16) * 1rem) + (18.67 - 14) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  background: #F15A28;
  border: 1px solid transparent;
}

.c0 svg {
  height: calc(((10.5 / 16) * 1rem) + (14.002500000000001 - 10.5) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
  width: calc(((14 / 16) * 1rem) + (18.67 - 14) * calc((min(100vw,2560px) - 320 / 16 * 1rem) / (2560 - 320)));
}

.c0:hover {
  background: #F26B3E;
  border: 1px solid transparent;
}

.c0:active {
  background: #F15A28;
  color: #FFFFFF;
}

<div>
    <button
      class="c0"
    >
      I am a button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
